### PR TITLE
[Inductor][CPP] Fix Mask Dtype mismatch

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142103

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/141559. The `vec_mask` store data type doesn't aligned when doing `bitwise_and`.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/cpp.py [hunk 1]
```diff
@@ -1269,6 +1269,9 @@ def logical_xor(a, b):
 
     @staticmethod
     def bitwise_and(a, b):
```
Comment (jgong5) [inline torch/_inductor/codegen/cpp.py:1271]: How about other related methods like `bitwise_or`?

Comment (leslie-fang-intel) [reply torch/_inductor/codegen/cpp.py:1271]: Extending the fix to
```
  bitwise_fns = [
      torch.bitwise_and,
      torch.bitwise_or,
      torch.bitwise_xor,
      torch.logical_and,
      torch.logical_or,
      torch.logical_xor,
  ]
```
Please help to take a look again.

**Review by jgong5 — APPROVED** [review]
